client-course-schedulizer/src/components/reuseables/Schedule/Schedule.tsx

```diff
@@ -1,37 +1,67 @@
 import { CalendarOptions } from "@fullcalendar/react";
-import React from "react";
+import { filter, forOwn } from "lodash";
+import React, { useContext } from "react";
 import Stick from "react-stick";
 import StickyNode from "react-stickynode";
-import { getHoursArr } from "../../../utilities/services/schedule";
+import { AppContext } from "../../../utilities/services/appContext";
+import { GroupedEvents, getHoursArr } from "../../../utilities/services/schedule";
+
 import { ScheduleToolbar } from "../../Toolbar/ScheduleToolbar";
 import { Calendar } from "../Calendar";
 import "./Schedule.scss";
 
 interface Schedule extends CalendarOptions {
   calendarHeaders: string[];
+  groupedEvents: GroupedEvents;
 }
 
 /* Creates a list of Calendars to create a Schedule
   <Stick> is used to stick the Schedule Header to the Schedule
   to track horizontal scrolling.
 */
-export const Schedule = ({ calendarHeaders, ...calendarOptions }: Schedule) => {
+export const Schedule = ({ calendarHeaders, groupedEvents, ...calendarOptions }: Schedule) => {
+  const {
+    appState: { selectedTerm, slotMaxTime, slotMinTime },
+  } = useContext(AppContext);
+
   const times = {
-    slotMaxTime: calendarOptions.slotMaxTime as string,
-    slotMinTime: calendarOptions.slotMinTime as string,
+    slotMaxTime,
+    slotMinTime,
+  };
+
+  calendarOptions = {
+    ...calendarOptions,
+    ...times,
   };
+
+  // Filter out events from other terms
+  forOwn(groupedEvents, (_, key) => {
+    groupedEvents[key] = filter(groupedEvents[key], (e) => {
+      return e.extendedProps?.section.term === selectedTerm;
+    });
+  });
+
+  // Filter out headers with no events
+  const calendarHeadersNoEmptyInTerm = filter(calendarHeaders, (header) => {
+    const groupEvents = groupedEvents[header];
+    return groupEvents?.length > 0;
+  });
```

Comment:
It might be worth doing research to see if it is important that we move logic out of the function component. I think these get redefined on every render, but maybe we put these in a useCallback()? Not sure. would be worth researching.

Reply:
I changed this to `useMemo`, but I'm unsure whether the dependencies do anything since they include objects and I'm worried that it does a reference comparison rather than a value comparison. If it uses reference comparison then I think the code would re-run on any re-render since an object would have a new location in memory even if it had the same values.

Reply:
I did some research though and it seems like it may not be worth worrying about anyway since the operation is not computationally expensive. React is fast enough that a re-render isn't a big deal